core/src/main/java/io/ddf/ds/User.java

```diff
@@ -0,0 +1,139 @@
+package io.ddf.ds;
+
+
+import com.google.common.base.Preconditions;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Represent a DDF user.
+ * Each user can keep a list of its own credentials to use on different source.
+ */
+public class User {
+
+  private static final ThreadLocal<User> currentUser = new ThreadLocal<>();
+
+  /**
+   * Get the current user in the system.
+   * The current user is keep on a per-thread basis.
+   *
+   * @return current user
+   */
+  public static User getCurrentUser() {
+    return currentUser.get();
+  }
+
+  /**
+   * Set the current user in the system.
+   * The current user is keep on a per-thread basis.
+   *
+   * @param user current user
+   */
+  public static void setCurrentUser(User user) {
+    currentUser.set(user);
+  }
+
+  private final String id;
+
+  private final String username;
+
+  private final Map<String, DataSourceCredential> authenticatedSources = new ConcurrentHashMap<>();
+
+  /**
+   * Create a new user with given username and a random id.
+   * @param username the username
+   */
+  public User(String username) {
+    this(UUID.randomUUID().toString(), username);
+  }
+
+  /**
+   * Create a new user with given id and username.
+   * @param id the user id
+   * @param username the username
+   */
+  public User(String id, String username) {
+    Preconditions.checkArgument(id != null, "id cannot be null");
+    Preconditions.checkArgument(username != null, "username cannot be null");
+
+    this.id = id;
+    this.username = username;
+  }
+
+  public String getId() {
+    return id;
+  }
+
+  public String getUsername() {
+    return username;
+  }
+
+  @Override
+  public String toString() {
+    return "User{" +
+        "id='" + id + '\'' +
+        ", username='" + username + '\'' +
+        '}';
+  }
+
+  @Override
+  public boolean equals(Object o) {
```

Comment:
@hai-adatao you mean for this equals method? It is pretty standard and was auto generated actually.